device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py

```diff
@@ -1,60 +1,331 @@
+                reg_file = open(qsfp_path, "r+")
+
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        content = reg_file.readline().rstrip()
+
+        # Absence of IOM throws read error
+        if (content == 'read error'):
+            return False
+
+        # content is a string containing the hex representation of the register
+        reg_value = int(content, 16)
+
+        # Rationalize port settings
+        if port_num >= self.iom2_port_start and port_num <= self.iom2_port_end:
+            port_num = port_num % 12
+	elif port_num >= self.iom3_port_start and port_num <= self.iom3_port_end:
+            port_num = port_num % 22
+
+        # Mask off the bit corresponding to our port
+        mask = (1 << port_num)
+        # LPMode is active high; set or clear the bit accordingly
+        if lpmode is True:
+            reg_value = reg_value | mask
+        else:
+            reg_value = reg_value & ~mask
+
+        # Convert our register value back to a hex string and write back
+        content = hex(reg_value)
+
+        reg_file.seek(0)
+        reg_file.write(content)
+        reg_file.close()
+
+        return True
+
+
+    def reset(self, port_num):
+
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+
+        # port_num and i2c match
+        if port_num >= self.iom1_port_start and port_num <= self.iom1_port_end:
+            i2c_line = 14
+        elif port_num >= self.iom2_port_start and port_num <= self.iom2_port_end:
+            i2c_line = 15
+	elif port_num >= self.iom3_port_start and port_num <= self.iom3_port_end:
+            i2c_line = 16
+
+        try:
+                qsfp_path = self.BASE_VAL_PATH.format(i2c_line)+"qsfp_lpmode"
+                reg_file = open(qsfp_path, "r+")
+
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        content = reg_file.readline().rstrip()
+
+        # File content is a string containing the hex representation of th
+        reg_value = int(content, 16)
+
+        # Rationalize port settings
+        if port_num >= self.iom2_port_start and port_num <= self.iom2_port_end:
+            port_num = port_num % 12
+	elif port_num >= self.iom3_port_start and port_num <= self.iom3_port_end:
+            port_num = port_num % 22
+
+        # Mask off the bit corresponding to our port
+        mask = (1 << port_num)
+
+        # ResetL is active low
+        reg_value = reg_value & ~mask
+
+        # Convert our register value back to a hex string and write back
+        reg_file.seek(0)
+        reg_file.write(hex(reg_value))
+        reg_file.close()
+
+        # Sleep 1 second to allow it to settle
+        time.sleep(1)
+
+        # Flip the bit back high and write back to the register to take
+	# port out of reset
```

Comment:
Replace tab with 4 spaces.

Reply:
Do we need to adhere "line too long (81 > 79 characters)" too. Because when I remove the spaces and realign the code I 
face these errors. Since most of the lines crossing the 79 limit sometimes throws some syntax issues..